client/src/auth.js

```diff
@@ -33,32 +33,42 @@ export class FakeStorage {
   removeItem(a) { return this._storage.delete(a) }
 }
 
-function getStorage() {
+function getStorage(storeLocally = false) {
+  let storage
   try {
-    if (typeof window !== 'object' || window.localStorage === undefined) {
-      return new FakeStorage()
+    if (!storeLocally ||
+        typeof window !== 'object' ||
+        window.localStorage === undefined) {
+      storage = new FakeStorage()
+    } else {
+      // Mobile safari in private browsing has a localStorage, but it
+      // has a size limit of 0
+      window.localStorage.setItem('$$fake', 1)
+      window.localStorage.removeItem('$$fake')
+      storage = window.localStorage
     }
-    // Mobile safari in private browsing has a localStorage, but it
-    // has a size limit of 0
-    window.localStorage.setItem('$$fake', 1)
-    window.localStorage.removeItem('$$fake')
-    return window.localStorage
   } catch (error) {
     if (window.sessionStorage === undefined) {
-      return new FakeStorage()
+      storage = new FakeStorage()
     } else {
-      return window.sessionStorage
+      storage = window.sessionStorage
     }
   }
+  return storage
 }
 
 export class TokenStorage {
   constructor({ authType = 'token',
-                storage = getStorage(),
+                storage = getStorage(authType.storeLocally),
```

Comment:
The intention is for authType.storeLocally to be falsey (undefined) if authType is not an object. If someone defines `storeLocally` to be true. I think I just messed up the logic for this one and it should be reversed.